### ISSUE TITLE
Direct Printing

### ISSUE THREAD

**Issue by mikeahrendsen** [frogmorecs/RawPrint#33]
We use the RawPrint assembly to do our direct printing of .pdf files and it works great.  However, we've had some issues pop up and was wondering if you could elaborate on them.

Printer Memory error.
One of our users has a 128mg memory on their printer and the direct print works fine, another user has a 64mg memory printer and the pdf output has the following message "PDF file not printed. 128 MB of memory is required to enable direct PDF printing".

PDF Contains Unreadable characters.
This user when doing direct prints finds that it prints “hundreds” of pages for a one page report with basically blank paper and only “garbage letters, numbers characters” at the top of a page.  This user has an older but I would think adequate printer.  They are using a Canon imageRunner Advance C5035, purchased in 2011.

Thank you in advance for sharing any insight on these items and thank you for providing this as a solution.  It has really helped and has gotten our application away from depending on a pdf reader.

Mike

**Comment by tonyedgecombe** [comment]
It does sound like that printer doesn't have enough memory, have you tried adding more?

The Canon printer doesn't support PostScript/PDF printing by default, it's an optional extra.

**Comment by mikeahrendsen** [comment]
Yes, adding more memory to the printer fixed this.  However, I only mentioned it because before we started using the RawPrint assembly the user wasn't getting this error, and was wondering if there was a memory restriction.

Thank you for the info on the Canon printer, I didn't realize that was an extra option.

**Comment by tonyedgecombe** [comment]
Presumably you were rendering the PDF on the host before you started using RawPrint. PDF files are complex, I'm not surprised they would need more resources on the printer than a typical print stream.
